pkg/services/datasources/service/datasource_service.go

```diff
@@ -303,7 +305,8 @@ func (s *Service) httpClientOptions(ds *models.DataSource) (*sdkhttpclient.Optio
 		}
 	}
 
-	if ds.JsonData != nil {
+	// Azure authentication
+	if ds.JsonData != nil && s.features.IsEnabled(featuremgmt.FlagHttpclientproviderAzureAuth) {
```

Comment:
Here you use a feature toggle and not in http_client_provider.go - can you explain why?

Reply:
I would want to completely remove ability to enable Azure authentication directly from `JsonData` and I would want to require all datasources to configure it programmatically via `opts.Azure`.

This code block behind the feature toggle still exists only because templating queries still require it.

More context here:
https://github.com/grafana/grafana-plugin-sdk-go/pull/468#issuecomment-1057622710

Also discussed here:
https://github.com/grafana/grafana/issues/37784#issuecomment-1030449903